lib/msprime.c

```diff
@@ -2116,6 +2119,35 @@ msp_common_ancestor_event(msp_t *self, population_id_t population_id)
 }
 
 
+/* This calculates the rate given by Eq (2) in the notes
+ */
+static double
+msp_compute_lambda_Xi_dirac(unsigned int b, double psi, double c)
```

Comment:
This function isn't really a "method" of the msp "class", since we don't pass a reference to the msp object in with it. All functions that have an ``msp_`` prefix should  be methods of the ``msp_`` class. There are two ways to look at this: (1) either we make it a method, and don't pass psi and c (which we should be able to get from the msp object state); or (2) make this an independent function, and remove the ``msp_`` prefix.